### ISSUE TITLE
Album cover art nor showing under "ALBUMS"

### ISSUE THREAD

**Issue by rompolompo** [abarisain/dmix#779]
I am using Nginx to serve images. All images are showing when I play or search for an artist. When using the "ALBUMS" tab, the list of the library comes up but no images. Clicking on "Reset Image" or "Download another image" does not work. 

Nexus 9 / Android 6.0
MPD 0.19.10

Screenshot:

![screenshot_20151126-232756](https://cloud.githubusercontent.com/assets/1028722/11434154/0ad0e2a8-9496-11e5-808b-682b9c6d72a6.png)

**Comment by avuton** [comment]
Pretty sure that's a restriction of the plugin. Can't look right now, but if I remember right there's a reason... And that's if that is the only enabled album art plugin.

**Comment by rompolompo** [comment]
I believe this used to work sometimes in the past. It was too buggy at that time and I have only started experimenting with this feature tonight. There was an issue with the "artist" and "albumartist" tags. Not sure if this is still the case.

**Comment by avuton** [comment]
Yes! There's a limitation with that screen in particular. We don't know the path for each of those albums, and that information would be very expensive to grab.

**Comment by avuton** [comment]
That's coming off my head. It'll work if you go to an artist first. It just will not function in that screen.

**Comment by avuton** [comment]
This could change in the future, but, I'm pretty sure it's never functioned like that. That would be very expensive.

**Comment by rompolompo** [comment]
Indeed. But you have to do this only once per album.

**Comment by avuton** [comment]
So, when you have an insane number of albums? That's a bad and arbitrary problem. Needs to be solved with a better solution. Either way, this will likely not be solved soon, if ever.

**Comment by rompolompo** [comment]
It is much easier to view your collection than to read the directories names.

**Comment by avuton** [comment]
Of course it is! Then stop limiting the album art to the local plugin, silly!

**Comment by rompolompo** [comment]
I have enabled "Download cover art". Same behavior.

**Comment by avuton** [comment]
Turn on the cache.

**Comment by rompolompo** [comment]
It was always on.

**Comment by avuton** [comment]
Works here. Requires a reset after changing some of these options.

**Comment by avuton** [comment]
Reset of the app. Kill-> restart

**Comment by avuton** [comment]
Which, also, you're saying cover downloading is completely broken? What version are you running?

**Comment by rompolompo** [comment]
Latest version. 1.08 Beta 4

**Comment by rompolompo** [comment]
Reset the app. Now there is only a list of albums. Not even the place holder for the cover art.

**Comment by avuton** [comment]
You're playing with the options. They're confusing. I suggest resetting them as I don't want to guide you through how to fix this. Though this does motivate me to work on removing some of this crap.

**Comment by rompolompo** [comment]
As you wish. We can try this later via Google Hangout if you have the time.

**Comment by abarisain** [comment]
>Now there is only a list of albums. Not even the place holder for the cover art. 

Yeah that happens when the cache is off. We will _never_ be able to get the cover art, so we show a list.

**Comment by hurzl** [comment]
As you don't have the artist shown I doubt you are using the cache ... ?

**Comment by rompolompo** [comment]
OK, took some time but I can reproduce this in the following scenario:

1. Remove "albumartist" from MPD "metadata to use"
2. Rescan my library
3. MPDroid does not load images under "ALBUMS"

1. Add "metadata to use"
2. Rescan my library
3. MPDroid does load images under "ALBUMS"

Also, the setting on MPDroid specifying which tag to use (artist, albumartist or both) seems irrelevant and have no impact on the result.

**Comment by abarisain** [comment]
What do your "albumartist" tags look like?

**Comment by rompolompo** [comment]
I rarely use it but when I do, it is the same as "artist"

**Comment by hurzl** [comment]
I have "metadata_to_use" undefined, so the default apparently contains the albumartist.

**Comment by rompolompo** [comment]
avuton, anything I can try to help with this?

**Comment by n03l** [comment]
I'm experiencing the same problem (with 1.08 beta 4). I tried changing the "metadata_to_use" (from albumartist to artist), but that didn't seem to help.

**Comment by n03l** [comment]
While playing around, I enable the _album cache_ (the _cover cache_ was already enabled), and now covers are showing up properly.
